internal/statestore/redis.go

```diff
@@ -447,84 +348,75 @@ func (rb *redisBackend) FilterTickets(ctx context.Context, pool *pb.Pool, pageSi
 	startTimeInt := curTime.Add(-ttl).UnixNano()
 
 	// Filter out tickets that are fetched but not assigned within ttl time (ms).
-	idsInIgnoreLists, err = redis.Strings(redisConn.Do("ZRANGEBYSCORE", "proposed_ticket_ids", startTimeInt, curTimeInt))
+	idsInIgnoreLists, err := redis.Strings(redisConn.Do("ZRANGEBYSCORE", "proposed_ticket_ids", startTimeInt, curTimeInt))
 	if err != nil {
 		redisLogger.WithError(err).Error("failed to get proposed tickets")
-		return status.Errorf(codes.Internal, err.Error())
+		return nil, status.Errorf(codes.Internal, "error getting ignore list %v", err)
+	}
+
+	idsIndexed, err := redis.Strings(redisConn.Do("SMEMBERS", allTickets))
+	if err != nil {
+		redisLogger.WithFields(logrus.Fields{
+			"Command": "SMEMBER allTickets",
+		}).WithError(err).Error("Failed to lookup all tickets.")
+		return nil, status.Errorf(codes.Internal, "error getting all indexed ticket ids %v", err)
 	}
 
-	// A set of playerIds that satisfies all filters
-	idSet := make([]string, 0)
+	r := make(map[string]struct{}, len(idsIndexed))
+	for _, id := range idsIndexed {
+		r[id] = struct{}{}
+	}
+	for _, id := range idsInIgnoreLists {
+		delete(r, id)
+	}
 
-	// For each filter, do a range query to Redis on Filter.DoubleArg
-	for i, filter := range extractIndexFilters(pool) {
-		// Time Complexity O(logN + M), where N is the number of elements in the DoubleArg set
-		// and M is the number of entries being returned.
-		// TODO: discuss if we need a LIMIT for # of queries being returned
-		idsInFilter, err = redis.Strings(redisConn.Do("ZRANGEBYSCORE", filter.name, filter.min, filter.max))
-		if err != nil {
-			redisLogger.WithFields(logrus.Fields{
-				"Command": fmt.Sprintf("ZRANGEBYSCORE %s %f %f", filter.name, filter.min, filter.max),
-			}).WithError(err).Error("Failed to lookup index.")
-			return status.Errorf(codes.Internal, "%v", err)
-		}
+	return r, nil
+}
 
-		if i == 0 {
-			idSet = idsInFilter
-		} else {
-			idSet = set.Intersection(idSet, idsInFilter)
-		}
+// GetTickets returns multiple tickets from storage.  Missing tickets are
+// silently ignored.
+func (rb *redisBackend) GetTickets(ctx context.Context, ids []string) ([]*pb.Ticket, error) {
```

Comment:
What instrumentation do we have for GetTickets? The performance for this method may be interesting as size / count of tickets goes up. I see this as the only real bottleneck in the new approach?

Reply:
Are you concerned that this will be introducing a new bottleneck, or that removing other bottlenecks only leaves this as the next remaining possible slowdown?
We already used mget in filter, and ticket query load on redis was far greater for the old system.

Reply:
I am not so much concerned about bottlenecks - as much as this is probably the most interesting data point now that we have eliminated the zranges due to query. If i look at this from what charts might be most interesting if we were to analyze scale for different ticket sizes, this one should be interesting. Need not do it as a part of this PR - but i think should be added as a part of the metrics we are adding anyways.

Reply:
Agreed that it'd be interesting, but is potential future work.